platform-api-client-gwt/che-core-client-gwt-factory/src/main/java/org/eclipse/che/api/factory/gwt/client/FactoryServiceClient.java

```diff
@@ -49,4 +51,11 @@
      * @param callback callback which returns snippet of the factory or exception if occurred
      */
     void getFactoryJson(@NotNull String workspaceId, @NotNull String path, @NotNull AsyncRequestCallback<Factory> callback);
+
+    /**
+     * Save factory to storage.
+     *
+     * @see FactoryService#saveFactory(Factory)
+     */
+    Promise<Factory> saveFactory(@NotNull Factory factory);
```

Comment:
we do not have specific rules on this, so as not to disrupt the overall design of the class I will not do it, and force this point for discussion